### PR TITLE
bpo-36239: Skip comments in gettext infos

### PR DESCRIPTION
**Pull request by JulienPalard**
https://bugs.python.org/issue36239

### CODE REVIEW

File: Lib/gettext.py [hunk 1]
```diff
@@ -417,6 +417,8 @@ def _parse(self, fp):
                     item = b_item.decode().strip()
                     if not item:
                         continue
+                    if item.startswith('#-#-#-#-#') and item.endswith('#-#-#-#-#'):
```
Comment (vstinner) [inline Lib/gettext.py:421]: Maybe add a comment saying something like: ignore "#-#-#-#-# ... #-#-#-#-#" comments.

File: Lib/test/test_gettext.py [hunk 2]
```diff
@@ -684,6 +684,13 @@ def test_plural_form_error_issue17898(self):
             # If this runs cleanly, the bug is fixed.
             t = gettext.GNUTranslations(fp)
 
+    def test_ignore_comments_in_headers_issue36239(self):
+        with open(MOFILE, 'wb') as fp:
+            fp.write(base64.decodebytes(GNU_MO_DATA_ISSUE_17898))
+        with open(MOFILE, 'rb') as fp:
+            t = gettext.GNUTranslations(fp)
+            self.assertEqual(t.info()["plural-forms"], "nplurals=2; plural=(n != 1);")
```
Comment (vstinner) [inline Lib/test/test_gettext.py:698]: Please add a comment explaining that the .mo file contains a comment. Something like:

Check that "#-#-#-#-#  messages.po (EdX Studio)  #-#-#-#-#" comment is ignored.

**Review by vstinner — APPROVED** [review]
LGTM, but IMHO the change deserves a few more comments :-)

File: Lib/gettext.py (hunk 1 above)
Comment (JulienPalard) [reply Lib/gettext.py:421]: did it.

File: Lib/test/test_gettext.py (hunk 2 above)
Comment (JulienPalard) [reply Lib/test/test_gettext.py:698]: good idea, done, thanks for proofreading Victor :)

File: Lib/gettext.py (hunk 1 above)
Comment (vstinner) [reply Lib/gettext.py:421]: Thanks.

**Pull request merged**